src/main/java/froggy/Deadline.java

```diff
@@ -0,0 +1,54 @@
+package froggy;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a task with description and a deadline.
+ */
+public class Deadline extends Task {
+
+    protected final String by;
+    protected LocalDate date;
```

Comment:
Can these have different names that are more descriptive? I see these in other places in your code.